File: temporal/multi-lang-demo/go-worker/shared.go
package main

type PaymentDetails struct {
	Amount        float64 `json:"amount"`
	SourceAccount string  `json:"sourceAccount"`
	TargetAccount string  `json:"targetAccount"`
	ReferenceID   string  `json:"referenceId"`
}

type ActivityResult struct {
	Success       bool   `json:"success"`
	TransactionId string `json:"transactionId"`
	Message       string `json:"message"`
}

const TaskQueueWithdrawal = "WITHDRAWAL_QUEUE"

File: temporal/multi-lang-demo/ts-worker/src/shared.ts
export interface PaymentDetails {
  amount: number;
  sourceAccount: string;
  targetAccount: string;
  referenceId: string;
}

export interface ActivityResult {
  success: boolean;
  transactionId: string;
  message: string;
}

export const TaskQueues = {
  MAIN_WORKFLOW: 'MULTI_LANG_TRANSFER_QUEUE',
  WITHDRAWAL: 'WITHDRAWAL_QUEUE',
  DEPOSIT: 'DEPOSIT_QUEUE',
  REFUND: 'REFUND_QUEUE',
} as const;